src/app/auth/components/registration/registration.component.html

```diff
@@ -0,0 +1,182 @@
+<form [formGroup]="registrationForm" class="regestration-form">
+  <h2>Registration</h2>
+  <!-- Email Field -->
+
+  <mat-form-field>
+    <mat-label>Enter your email</mat-label>
+    <input matInput placeholder="example@email.com" formControlName="email" />
+    @if (registrationForm.get("email")?.hasError("required")) {
+      <mat-error>You must enter an email</mat-error>
+    }
+    @if (registrationForm.get("email")?.hasError("email")) {
+      <mat-error>Provide valid email</mat-error>
+    }
+  </mat-form-field>
+
+  <!-- Password Fields -->
+
+  <mat-form-field>
+    <mat-label>Enter your password</mat-label>
+    <input
+      matInput
+      formControlName="password"
+      [type]="hidepassword ? 'password' : 'text'"
+    />
+    <button
+      mat-icon-button
+      matSuffix
+      (click)="hidepassword = !hidepassword"
+      [attr.aria-label]="'Hide password'"
+      [attr.aria-pressed]="hidepassword"
+    >
+      <mat-icon>{{ hidepassword ? "visibility_off" : "visibility" }}</mat-icon>
+    </button>
+    @if (registrationForm.get("password")?.hasError("noLowerCase")) {
+      <mat-error>Password must include at least one lowercase letter</mat-error>
+    }
+    @if (registrationForm.get("password")?.hasError("noUpperCase")) {
+      <mat-error>Password must include at least one uppercase letter</mat-error>
+    }
+    @if (registrationForm.get("password")?.hasError("noDigit")) {
+      <mat-error>Password must include at least one digit</mat-error>
+    }
+    @if (registrationForm.get("password")?.hasError("noSimbol")) {
+      <mat-error
+        >Password must include at least one special character
+        (&#33;&#64;&#35;&#63;)</mat-error
+      >
+    }
+    @if (registrationForm.get("password")?.hasError("tooShort")) {
+      <mat-error>Password must be at least 8 characters long</mat-error>
+    }
+  </mat-form-field>
+
+  <mat-form-field appearance="fill">
+    <mat-label>Confirm your password</mat-label>
+    <input
+      matInput
+      formControlName="confirmPassword"
+      [type]="hidepassword ? 'password' : 'text'"
+    />
+    @if (registrationForm.get("confirmPassword")?.hasError("required")) {
+      <mat-error>You need to confirm password</mat-error>
+    }
+
+    @if (registrationForm.errors?.["PasswordNoMatch"]) {
+      <mat-error>Passwords do not match</mat-error>
+    }
+    @if (registrationForm.get("confirmPassword")?.hasError("matching")) {
+      <mat-error>Passwords do not match</mat-error>
+    }
+  </mat-form-field>
+  <!-- name Field -->
+
+  <mat-form-field>
+    <mat-label>First name</mat-label>
+    <input
+      matInput
+      formControlName="firstName"
+      placeholder="Enter your first name"
+    />
+    @if (registrationForm.get("firstName")?.hasError("required")) {
+      <mat-error>Name is required</mat-error>
```

Comment:
Message texts and information that is not dynamic from BE, but can be reusable is better to have in constant.